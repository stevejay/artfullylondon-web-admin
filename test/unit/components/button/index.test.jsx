import React from 'react'
import { shallow } from 'enzyme'
import _ from 'lodash'

import Button from '_admin/components/button'

it('should render correctly', () => {
  const wrapper = shallow(<Button onClick={_.noop}><div id='child' /></Button>)
  expect(wrapper).toMatchSnapshot()
})

it('should render correctly when submitting', () => {
  const wrapper = shallow(
    <Button onClick={_.noop} submitting><div id='child' /></Button>
  )

  expect(wrapper).toMatchSnapshot()
})

it('should render correctly when disabled', () => {
  const wrapper = shallow(
    <Button onClick={_.noop} disabled><div id='child' /></Button>
  )

  expect(wrapper).toMatchSnapshot()
})
