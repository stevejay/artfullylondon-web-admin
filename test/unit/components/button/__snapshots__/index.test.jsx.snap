// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<button
  className="components-button-index-m__default components-button-index-m__button"
  onClick={[Function]}
  type="button"
>
  <span
    className="components-button-index-m__content"
  >
    <div
      id="child"
    />
  </span>
</button>
`;

exports[`should render correctly when disabled 1`] = `
<button
  className="components-button-index-m__disabled components-button-index-m__button"
  disabled={true}
  onClick={[Function]}
  type="button"
>
  <span
    className="components-button-index-m__content"
  >
    <div
      id="child"
    />
  </span>
</button>
`;

exports[`should render correctly when submitting 1`] = `
<button
  className="components-button-index-m__submitting components-button-index-m__button"
  onClick={[Function]}
  type="button"
>
  <span
    className="components-button-index-m__content"
  >
    <div
      id="child"
    />
  </span>
  <Loader
    size="medium"
    type="inverse"
  />
</button>
`;
